stream-chat-android-client/src/main/java/io/getstream/chat/android/client/StreamLifecycleObserver.kt

```diff
@@ -10,19 +10,26 @@ import kotlinx.coroutines.launch
 
 internal class StreamLifecycleObserver(private val handler: LifecycleHandler) : LifecycleObserver {
     private var recurringResumeEvent = false
+    @Volatile private var isObserving = false
 
     fun observe() {
-        CoroutineScope(DispatcherProvider.Main).launch {
-            ProcessLifecycleOwner.get()
-                .lifecycle
-                .addObserver(this@StreamLifecycleObserver)
+        if (isObserving.not()) {
+            isObserving = true
+            CoroutineScope(DispatcherProvider.Main).launch {
```

Comment:
Maybe it's good to store scope as a local private property?

Reply:
DONE ;)